View individual event

cheljoh opened the pull request:
Add functionality for viewing specific events. Group tickets for listing. Presently, seat is not presented in view, only row. May need to change in future.

cheljoh committed:
Start Event Show

Started event show, need to implement listing functionality on event show.

cheljoh committed:
User Views Individual Events

Added functionality for viewing specific events. Display tickets by listing.
Row is present in panel, but not individual ticket seat. May change in future.

cheljoh commented on db/seeds.rb:
```diff
@@ -35,13 +36,11 @@
+concerts.events.create(
+  name: "Muppet Rock",
+  venue_id: venue2.id,
+  image: open("https://s3.amazonaws.com/ticket-cacher/categories/muppets_rocking.jpg"),
```
Will ignore hound this time, don't want to split up URL.

cheljoh committed:
added model methods to clean up views

cheljoh committed:
Merge branch 'master' into view_individual_event

cheljoh committed:
Add Links

Add category links to navbar.

erinnachen commented on spec/features/user/user_views_events_by_category_spec.rb:
```diff
@@ -8,9 +8,9 @@
+    visit "/festivals"
 
-    click_on "festivalz"
+    # add css image
```
is there something you're missing here?

erinnachen commented on spec/features/user/user_views_a_specific_event_spec.rb:
```diff
@@ -0,0 +1,38 @@
+      expect(page).to have_content("row: 7")
+      expect(page).to have_content("price: $10")
+      expect(page).to have_content("quantity: 1")
+    end
```
maybe add this: expect(page).to_not have_content("quantity: 2")

erinnachen commented on app/views/shared/_user_functions.html.erb:
```diff
@@ -5,25 +5,22 @@
-    </li>
-    <li class="name-in-dropdown">
-      <%= link_to "comedy", "#" %>
+      <%= link_to "comedy", "comedy" %>
```
should there be a slash here ("/comedy")

erinnachen commented on app/models/event.rb:
```diff
@@ -1,6 +1,8 @@
   belongs_to :category
   belongs_to :venue
+  has_many :listings
+
```
is this relationship tested?

erinnachen commented on app/models/event.rb:
```diff
@@ -1,6 +1,8 @@
+
   # has_many :order_products
   # has_many :orders, through: :order_products
 
```
we should take out these comments

cheljoh committed:
Add model tests

Added shoulda-matcher model tests to test relationship.
Also changed feature test for viewing individual test to make it more robust.

damwhit commented:
I like it.  Glad you went back to test more models.
